public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -274,36 +274,47 @@ export class PanelChrome extends Component<Props, State> {
     const headerHeight = this.hasOverlayHeader() ? 0 : theme.panelHeaderHeight;
     const chromePadding = plugin.noPadding ? 0 : theme.panelPadding;
     const panelWidth = width - chromePadding * 2 - PANEL_BORDER;
+    //const scrollableYMargin = plugin.scrollable ? chromePadding : 0;
     const innerPanelHeight = height - headerHeight - chromePadding * 2 - PANEL_BORDER;
-    const panelContentClassNames = classNames({
-      'panel-content': true,
-      'panel-content--no-padding': plugin.noPadding,
-    });
+
+    const contentStyles: CSSProperties = {
+      padding: chromePadding,
+      width: '100%',
+      flexGrow: 1,
+      height: `calc(100% - ${headerHeight}px)`,
+      overflow: 'hidden',
+    };
+
+    if (plugin.scrollable) {
+      contentStyles.paddingRight = '0';
+    }
+
     const panelOptions = panel.getOptions();
+    const panelComponent = (
+      <PanelComponent
+        id={panel.id}
+        data={data}
+        title={panel.title}
+        timeRange={timeRange}
+        timeZone={this.props.dashboard.getTimezone()}
+        options={panelOptions}
+        fieldConfig={panel.fieldConfig}
+        transparent={panel.transparent}
+        width={panelWidth}
+        height={innerPanelHeight}
+        renderCounter={renderCounter}
+        replaceVariables={panel.replaceVariables}
+        onOptionsChange={this.onOptionsChange}
+        onFieldConfigChange={this.onFieldConfigChange}
+        onChangeTimeRange={this.onChangeTimeRange}
+        eventBus={dashboard.events}
+      />
+    );
 
     return (
-      <>
-        <div className={panelContentClassNames}>
-          <PanelComponent
-            id={panel.id}
-            data={data}
-            title={panel.title}
-            timeRange={timeRange}
-            timeZone={this.props.dashboard.getTimezone()}
-            options={panelOptions}
-            fieldConfig={panel.fieldConfig}
-            transparent={panel.transparent}
-            width={panelWidth}
-            height={innerPanelHeight}
-            renderCounter={renderCounter}
-            replaceVariables={panel.replaceVariables}
-            onOptionsChange={this.onOptionsChange}
-            onFieldConfigChange={this.onFieldConfigChange}
-            onChangeTimeRange={this.onChangeTimeRange}
-            eventBus={dashboard.events}
-          />
-        </div>
-      </>
+      <div className="panel-content" style={contentStyles}>
+        {plugin.scrollable ? <CustomScrollbar autoHeightMin="100%">{panelComponent}</CustomScrollbar> : panelComponent}
```

Comment:
the main blocker that I was unable to fix was that panels that use the setScrollable now lacks a right padding (see it in the dashboard list panel). Not easy to solve. One way would be to customize the scrollbar inner view styles to add the padding back there.

Reply:
The bar gauge does not have this as it respects the width property sent to it the panel (but dash list panel just has full width cards)